app/v1/services/RetrieveTransactionDetailsService.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.services
+
+import cats.data.EitherT
+import cats.implicits._
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging
+import v1.connectors.RetrieveTransactionDetailsConnector
+import v1.controllers.EndpointLogContext
+import v1.models.errors._
+import v1.models.outcomes.ResponseWrapper
+import v1.models.request.retrieveTransactionDetails.RetrieveTransactionDetailsParsedRequest
+import v1.models.response.retrieveTransactionDetails.RetrieveTransactionDetailsResponse
+import v1.support.DesResponseMappingSupport
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveTransactionDetailsService @Inject()(val connector: RetrieveTransactionDetailsConnector) extends DesResponseMappingSupport with Logging {
+
+  def retrieveDetails(request: RetrieveTransactionDetailsParsedRequest)(
```

Comment:
I would prefer if this was either 'retrieveTransactionDetails' or just 'retrieve' as the full name of the allocations endpoint is 'retrieve a self assessment payment's allocation details' so this could potentially be ambiguous.